ffi/koptcontext.lua

```diff
@@ -117,7 +111,7 @@ function KOPTContext_mt.__index:getWordBoxes(bmp, x, y, w, h, box_type)
         counter_cw = 0
         l_x0, l_y0, l_x1, l_y1 = 9999, 9999, 0, 0
         while current_line == counter_l[0] and counter_w < nr_word do
-            box = leptonica.boxaGetBox(boxa, counter_w, ffi.C.L_CLONE)
+            local box = leptonica.boxaGetBox(boxa, counter_w, ffi.C.L_CLONE)
```

Comment:
If it's the same box that you remove the ffi declaration above, better add it back just there:
```lua
       local box = ffi.new("BOX[1]")
       box = leptonica.boxaGetBox(boxa, counter_w, ffi.C.L_CLONE)
```

Reply:
It seems redundant. My own testing seemed fine, the unit tests didn't break,[1] and elsewhere in the code it doesn't do that either.[2] :-/

[1]
https://github.com/koreader/koreader-base/blob/b73bfe5f8b9e94c27d8af6272ae39a536347dc96/spec/unit/koptcontext_spec.lua#L86-L109

[2]
https://github.com/koreader/koreader-base/blob/f8caa6ab3d1eb8fa0e70d2232082f2957c418a0d/ffi/koptcontext.lua#L241
https://github.com/koreader/koreader-base/blob/f8caa6ab3d1eb8fa0e70d2232082f2957c418a0d/ffi/koptcontext.lua#L269

Reply:
OK then (some experiments of mine with python ctypes (the equivalent of lua ffi) lead me to think that it's always best - to avoid segfaults - to declare the ctype type of the returned value - may be lua ffi is better with that)

Reply:
OK, I guess it's the necessary include of things like require("ffi/leptonica_h") that helps ffi to make the right lua object as it knows what type of things a function returns.

Reply:
Lua is definitely better than Python (in such matters).